### PR TITLE
Started work on freeing a node

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: doc/Code/Juvix.org [hunk 1]
```diff
@@ -58,6 +58,39 @@ This module provides the default environment to run LLVM code in
...
 Operations necessary to update nodes
+- Allocation layout as described by =mallocNodeH=
+  + layout :
+    [portSize | PortArray[portLocation | NodePtr] | DataArray[Data]]
+  | Part         | Alloca Or Malloc  |
+  |--------------+-------------------|
+  | portSize     | Malloc            |
+  | PortArray    | Malloc            |
+  | DataArray    | Malloc            |
+  | PortLocation | Previously Allocd |
```
Comment (cwgoes) [inline doc/Code/Juvix.org]: Can we document where?

File: doc/Code/Juvix.org [hunk 2]
```diff
@@ -99,6 +132,22 @@ Provides a mechanism for defining Sum types
...
+  + layout :
+    {eac | NodePtr* [portSize | PortArray[portLocation | NodePtr] | DataArray[Data]]}
+    * Similar to the one in Graph, however it also has the eac tag
+   | Part         | Alloca Or Malloc   |
+   |--------------+--------------------|
+   | eac          | Malloc             |
+   | portSize     | Malloc             |
+   | PortArray    | Malloc             |
+   | DataArray    | Malloc             |
+   | PortLocation | Not Allocated Here |
```
Comment (cwgoes) [inline doc/Code/Juvix.org]: Can we document where (ditto for `NodePtr`, `Data`)?

File: src/Juvix/Backends/LLVM/Codegen/Graph.hs [hunk 3]
```diff
@@ -259,7 +295,31 @@ defineRewire nodePtrType = Block.defineFunction Type.void "rewire" args $
         (numPorts, "port_two")
       ]
 
-delNode = undefined
+deAllocateNode ∷ Define m ⇒ Operand.Operand → Type.Type → m Operand.Operand
+deAllocateNode nodePtr nodePtrType = do
+  node ← load (nodeType nodePtrType) nodePtr
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Codegen/Graph.hs]: So, reading this I am sorta wondering why these aren't contiguous

If they were contiguous (even if they had separate pointers) we could free them with one `free`, one would suppose.

We can worry about this later though

**Comment by cwgoes** [comment]
If we combine all the docs, we should end up with a table which has a bidirectional mapping between every structure stored in memory and the associated `malloc` / `free` calls for that structure

File: doc/Code/Juvix.org (hunk 1 above)
Comment (mariari) [reply doc/Code/Juvix.org]: Will add tomorrow!

File: src/Juvix/Backends/LLVM/Codegen/Graph.hs (hunk 3 above)
Comment (mariari) [reply src/Juvix/Backends/LLVM/Codegen/Graph.hs]: I'd say let's make a TODO of this, I actually thought of this myself after writing it.

However, I decided to can it for the time being, as I'm not the most confident in the free and malloc behaviors when interacting with LLVM code.

File: src/Juvix/Backends/LLVM/Codegen/Graph.hs [hunk 4]
```diff
@@ -259,7 +295,31 @@ defineRewire nodePtrType = Block.defineFunction Type.void "rewire" args $
         (numPorts, "port_two")
       ]
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Codegen/Graph.hs]:
```suggestion
-- TODO: Should these be contiguous? If they were contiguous (even if they had separate pointers) we could free them with one free, one would suppose.
```

File: src/Juvix/Backends/LLVM/Codegen/Graph.hs (hunk 3 above)
Comment (cwgoes) [reply src/Juvix/Backends/LLVM/Codegen/Graph.hs]: Added a TODO

**Pull request merged**